File: SPCode/Reporting/Support/Interfaces.vb
Namespace Reporting
    Public Class SPReportFilterValue

        Private _Name As String
        Private _Value As Object = Nothing

        Public Property Name() As String
            Get
                Return _Name
            End Get
            Set(ByVal value As String)
                _Name = value
            End Set
        End Property
        Public Property Value() As Object
            Get
                Return _Value
            End Get
            Set(ByVal value As Object)
                _Value = value
            End Set
        End Property

    End Class

    Public Class SPReportFilterValueCollection
        Inherits List(Of SPReportFilterValue)

    End Class

    Public Delegate Function SPAdvancedSearchDelegate(ByVal Options As String) As SPReportFilterValue
    Public Delegate Function SPFilterSetupDelegate(ByVal Options As String) As SPReportFilterValueCollection

End Namespace

File: SPCode/CS/Reporting/Support/Interfaces.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;

namespace SoftLogik.Win
{
	namespace Reporting
	{
		public class SPReportFilterValue
		{
			
			
			private string _Name;
			private object _Value = null;
			
			public string Name
			{
				get
				{
					return _Name;
				}
				set
				{
					_Name = value;
				}
			}
			public object Value
			{
				get
				{
					return _Value;
				}
				set
				{
					_Value = value;
				}
			}
			
		}
		
		public class SPReportFilterValueCollection : List<SPReportFilterValue>
		{
			
			
		}
		
		public delegate SPReportFilterValue  SPAdvancedSearchDelegate(string Options);
		public delegate SPReportFilterValueCollection  SPFilterSetupDelegate(string Options);
		
	}
	
}
